pkg/types/installconfig.go

```diff
@@ -47,11 +47,14 @@ type InstallConfig struct {
 	// Networking defines the pod network provider in the cluster.
 	*Networking `json:"networking,omitempty"`
 
-	// Machines is the list of MachinePools that need to be installed.
+	// ControlPlane is the configuration for the machines that comprise the
+	// control plane.
 	// +optional
-	// Default on AWS and OpenStack is 3 masters and 3 workers.
-	// Default on Libvirt is 1 master and 1 worker.
-	Machines []MachinePool `json:"machines,omitempty"`
+	ControlPlane *MachinePool `json:"controlPlane,omitempty"`
```

Comment:
Why is this omitempty? and nilable.

from the commit message **`There should always be a control plane machine pool`**

Reply:
Because the installer can fill in the default.

Reply:
It is omit empty and nilable because the user does not have to provide it. If it is omitted, then the installer will add default values to the InstallConfig used. The validator will ensure that the field is not nil in the final InstallConfig.

Reply:
:confused: sure